crates/ruff_linter/src/rules/flake8_pyi/rules/old_style_positional_only_arg.rs

```diff
@@ -0,0 +1,84 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::ParameterWithDefault;
+use ruff_python_semantic::{analyze::function_type, Definition};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for the presence of old-style positional-only arguments in stub files.
+///
+/// ## Why is this bad?
+/// [PEP 570][1] defines new syntax for positional-only arguments, that should
+/// be preferred over the syntax defined in [PEP 484][2].
+///
+/// ## Example
+/// ```python
+/// def foo(__x: int) -> None:
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def foo(x: int, /) -> None: ...
+/// ```
+///
+/// [1]: https://peps.python.org/pep-0570
+/// [2]: https://peps.python.org/pep-0484/#positional-only-arguments
+#[violation]
+pub struct OldStylePositionalOnlyArg;
+
+impl Violation for OldStylePositionalOnlyArg {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Prefer PEP 570 syntax for positional-only arguments in stubs")
+    }
+}
+
+/// PYI063
+pub(crate) fn old_style_positional_only_arg(checker: &mut Checker, definition: &Definition) {
+    let Some(function) = definition.as_function_def() else {
+        return;
+    };
+    if !function.parameters.posonlyargs.is_empty() {
+        return;
+    }
+
+    let mut args = function.parameters.args.iter();
+    let Some(first_arg) = args.next() else {
+        return;
+    };
+
+    let semantic = checker.semantic();
+    // TODO: this scope is wrong.
+    let scope = semantic.current_scope();
+    let function_type = function_type::classify(
```

Comment:
This call currently doesn't correctly identify methods, which is why the issue is not correctly raised on line 26 of `PYI063.pyi`.

I didn't look super hard into it, but I don't see a very straightforward way to get the correct scope from a `FunctionStmt` node.

Reply:
I fixed this by moving it out of the definition phase and into the standard statement checker.